src/submission/submission-service.ts

```diff
@@ -641,6 +646,72 @@ export namespace operations {
     );
   };
 
+  export async function getAllCommittedClinicalData(programId: string) {
+    if (!programId) throw new Error('Missing programId!');
+
+    // this query takes about 6sec for 20,000 donors with sample registration only running locally
+    const donors = await getDonors(programId); // get in batch??
+
+    // collect all records
+    const recordsMap: any = {};
+    donors.forEach(d => {
+      Object.values(ClinicalEntitySchemaNames).forEach(entity => {
+        let clincialInfoRecords;
+        if (entity === ClinicalEntitySchemaNames.REGISTRATION) {
+          clincialInfoRecords = generateSampleRegistrationRecordsFromDonor(d);
+        } else {
+          clincialInfoRecords = getClinicalEntitiesFromDonorBySchemaName(d, entity);
+        }
+        recordsMap[entity] = _.concat(recordsMap[entity] || [], clincialInfoRecords);
+      });
+    });
+
+    // map into object ready for api processing
+    const schemasWithFields = schemaManager.instance().getSchemasWithFields();
+    const schemaVersion = schemaManager.instance().getCurrent().version;
+    return Object.entries(recordsMap)
+      .map(([schemaEntityName, records]) => {
+        if (isEmpty(records)) return undefined;
+
+        const relevantSchemaWithFields = schemasWithFields.find(s => s.name === schemaEntityName);
+        if (!relevantSchemaWithFields) {
+          throw new Error(`Can't find schema ${schemaEntityName}, shouldn't be possbile`);
+        }
+
+        return {
+          schemaEntityName: relevantSchemaWithFields.name,
+          schemaVersion,
```

Comment:
different donors can be on different versions

Reply:
So I did think about this a bit. But the donor version doesn't imply the clinical record is on the same version. Like they could have submitted and verified bu the donor schema version could be old

Reply:
Honestly I think this [ticket](https://github.com/icgc-argo/platform-ui/issues/1318) is misleading cause it feels more like a give me recrods that I can resubmit which is why I try to map them to the headers/fieldsNames in current schema